File: Server/app/schemas/quiz.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class SibiQuizQuestionResponse(BaseModel):
    id: int
    question_text: str
    image_path: Optional[str] = None
    options: List[str]
    correct_answer: str

    class Config:
        from_attributes = True

class SibiQuizSubmitRequest(BaseModel):
    score: int
    correct_answers: int
    total_questions: int

class SibiQuizHistoryResponse(BaseModel):
    id: int
    score: int
    correct_answers: int
    total_questions: int
    created_at: datetime

    class Config:
        from_attributes = True

File: Client/src/types/quiz.ts
export interface QuizQuestion {
  id: number;
  question_text: string;
  image_path: string | null;
  options: string[];
  correct_answer: string;
}

export interface QuizHistoryItem {
  id: number;
  score: number;
  correct_answers: number;
  total_questions: number;
  created_at: string;
}

export interface QuizLobbyProps {
  loading: boolean;
  error: string | null;
  startQuiz: () => void;
  loadingHistory: boolean;
  history: QuizHistoryItem[];
}

export interface QuizActiveProps {
  currentIdx: number;
  questions: QuizQuestion[];
  selectedAns: string | null;
  handleAnswerSelect: (option: string) => void;
  API_BASE_URL: string;
}

export interface QuizFinishedProps {
  correctCount: number;
  totalQuestions: number;
  startQuiz: () => void;
}

export interface InfoCardProps {
    number: string;
    title: string;
    description: string;
}